gemfirexd/core/src/main/javacc/com/pivotal/gemfirexd/internal/impl/sql/compile/sqlgrammar.jj

```diff
@@ -12277,8 +12277,16 @@ partitioningClause(DistributionDefinitionNode distributionNode, boolean[] repPar
 	PartitionResolver tmpprslvr = null;
 }
 {
-	<PARTITION> <BY> 
-	(  
+	<PARTITION> <BY>
+
+  	(
+  	 <HASH>
+  	 {
+  	     distributionNode.setCustomHashing(true);
+  	  }
+  	)?
+	(
```

Comment:
Perhaps we can name it like SPARKHASH or similar to make intent clear? (a non-reserved keyword)